packages/shared/src/hooks/contentPreference/useContentPreference.ts

```diff
@@ -152,10 +156,72 @@ export const useContentPreference = (): UseContentPreference => {
     },
   });
 
+  const { mutateAsync: block } = useMutation({
+    mutationKey: generateQueryKey(RequestKey.ContentPreferenceBlock, user),
+    mutationFn: async ({
+      id,
+      entity,
+      entityName,
+      opts,
+    }: PropsParameters<UseContentPreference['block']>) => {
+      if (!user) {
+        showLogin({ trigger: AuthTriggers.Follow });
+
+        throw new Error('not logged in');
+      }
+
+      logEvent({
+        event_name: LogEvent.Block,
+        target_id: id,
+        target_type: entityName,
+        extra: opts?.extra ? JSON.stringify(opts.extra) : undefined,
+      });
+
+      await gqlClient.request(CONTENT_PREFERENCE_BLOCK_MUTATION, {
+        id,
+        entity,
+      });
+
+      displayToast(`⛔️ You blocked the following words: ${entityName}`);
```

Comment:
`entityName` would be `word` and then `id` would be what you blocked no? Should be generic message so we can use `entityName` instead of hardcoded `words`,

Reply:
If you block user message should also work.

Reply:
Yeah will adjust a bit :)